armi/nucDirectory/elements.py

```diff
@@ -25,124 +27,155 @@
 
+class ChemicalPhase(Enum):
+    GAS = 1
+    LIQUID = 2
+    SOLID = 3
+    UNKNOWN = 4
 
-class Element:
-    r"""
-    Represents an element, defined by its atomic number.
 
-    Attributes
-    ----------
-    z : int
-        atomic number, number of protons
+class ChemicalGroup(Enum):
+    ALKALI_METAL = 1
+    ALKALINE_EARTH_METAL = 2
+    NONMETAL = 3
+    TRANSITION_METAL = 4
+    POST_TRANSITION_METAL = 5
+    METALLOID = 6
+    HALOGEN = 7
+    NOBEL_GAS = 8
+    LANTHANIDE = 9
+    ACTINIDE = 10
+    UNKNOWN = 11
 
-    symbol : str
-        element symbol
 
-    name : str
-        element name
-
-    nuclideBases : list of nuclideBases
-        nuclideBases for this element
-    """
+class Element:
+    """Represents an element defined on the Periodic Table."""
 
-    def __init__(self, z, symbol, name):
-        r"""
+    def __init__(self, z, symbol, name, phase="UNKNOWN", group="UNKNOWN"):
+        """
         Creates an instance of an Element.
 
         Parameters
         ----------
         z : int
             atomic number, number of protons
-
         symbol : str
             element symbol
-
         name: str
             element name
-
+        phase : str
+            Chemical phase of the element at standard temperature and pressure (e.g., gas, liquid, solid).
+        group : str
+            Chemical group of the element.
         """
         self.z = z
         self.symbol = symbol
         self.name = name
+        self.phase = ChemicalPhase[phase]
+        self.group = ChemicalGroup[group]
         self.standardWeight = None
-        self.nuclideBases = []
-
-        other = byZ.get(z, None)
-        if other is not None and other == self:
-            raise Exception(
-                "Element with atomic weight {} already exists" "".format(self)
-            )
-        byZ[z] = self
-        byName[name] = self
-        bySymbol[symbol] = self
+        self.nuclides = []
+        addGlobalElement(self)
 
     def __repr__(self):
-        return "<Element {} {}>".format(self.symbol, self.z)
+        return f"<Element {self.symbol:>3s} (Z={self.z}), {self.name}, {self.group}, {self.phase}>"
 
     def __eq__(self, other):
         return (
             self.z == other.z
             and self.symbol == other.symbol
             and self.name == other.name
+            and self.phase == other.phase
+            and self.group == other.group
+            and len(self.nuclides) == len(other.nuclides)
         )
 
     def __hash__(self):
         return hash(self.name)
 
     def __iter__(self):
-        for nuc in self.nuclideBases:
+        for nuc in sorted(self.nuclides):
             yield nuc
 
     def append(self, nuclide):
-        self.nuclideBases.append(nuclide)
+        """Assigns and sorts the nuclide to the element and ensures no duplicates."""
+        from armi.nucDirectory.nuclideBases import NuclideBase
+
+        if nuclide in self.nuclides or not isinstance(nuclide, NuclideBase):
```

Comment:
I understand just returning if the nuclide is already in the list, but do you think we should instead fail if the `nuclide` isn't a valid instance?